kalite/topic_tools/content_models.py

```diff
@@ -395,11 +396,21 @@ def annotate_content_models(db=None, channel="khan", language="en", ids=None, **
     Our ids can be duplicated at the moment, so this may be several content items per id.
     When a content item has been updated, propagate availability up the topic tree.
     """
-    content_models = get_content_items(ids=ids, channel=channel, language=language)
-    updates_dict = update_content_availability(content_models)
 
-    with Using(db, [Item]):
+    def iterator_content_items(ids=None, **kwargs):
+        if ids:
+            items = Item.select().where(Item.id.in_(ids)).dicts().iterator()
+        else:
+            items = Item.select().dicts().iterator()
+
+        mapped_items = itertools.imap(unparse_model_data, items)
+        updated_mapped_items = update_content_availability(mapped_items)
 
+        for path, update in updated_mapped_items:
+            yield path, update
+
+    with Using(db, [Item]):
+        content_models = iterator_content_items(ids=ids, channel=channel, language=language)
         with db.atomic() as transaction:
```

Comment:
With this whole block as an atomic transaction, what are the memory implications?